server/src/main/java/org/apache/druid/catalog/model/MeasureTypes.java

```diff
@@ -0,0 +1,219 @@
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+public class MeasureTypes
+{
+  public static final String OPTIONAL = "--";
+
+  public enum BaseType
+  {
+    VARCHAR(ColumnType.STRING),
+    BIGINT(ColumnType.LONG),
+    FLOAT(ColumnType.FLOAT),
+    DOUBLE(ColumnType.DOUBLE);
+
+    public final ColumnType nativeType;
+
+    BaseType(ColumnType nativeType)
+    {
+      this.nativeType = nativeType;
+    }
+  }
+
+  public static class MeasureType
+  {
+    public final String name;
+    public final List<BaseType> argTypes;
+    public final String sqlSeedFn;
+    public final String sqlReducerFn;
+    public final String nativeType;
+    public final String nativeAggFn;
+    public final ColumnType storageType;
+    public final String nativeReducerFn;
+
+    public MeasureType(
+        final String name,
+        final List<BaseType> argTypes,
+        final String sqlSeedFn,
+        final String sqlReducerFn,
+        final String nativeType,
+        final ColumnType storageType,
+        final String nativeAggFn,
+        final String nativeReducerFn
+    )
+    {
+      this.name = name;
+      this.argTypes = argTypes == null ? Collections.emptyList() : argTypes;
+      this.sqlSeedFn = sqlSeedFn;
+      this.sqlReducerFn = sqlReducerFn;
+      this.nativeType = nativeType;
+      this.storageType = storageType;
+      this.nativeAggFn = nativeAggFn;
+      this.nativeReducerFn = nativeReducerFn;
+    }
+
+    @Override
+    public String toString()
+    {
+      StringBuilder buf = new StringBuilder()
+          .append(name)
+          .append("(");
+      for (int i = 0; i < argTypes.size(); i++) {
+        if (i > 0) {
+          buf.append(", ");
+        }
+        buf.append(argTypes.get(i).name());
+      }
+      return buf.append(")").toString();
+    }
+  }
+
+  // See: https://druid.apache.org/docs/latest/querying/aggregations.html
+  public static final MeasureType COUNT_TYPE = new MeasureType(
+      "COUNT",
+      null,
+      null,
+      "SUM",
+      "longSum",
+      ColumnType.LONG,
+      "count",
+      "longSum"
+  );
+
+  public static final MeasureType SUM_BIGINT_TYPE = simpleAggType("sum", BaseType.BIGINT);
+  public static final MeasureType SUM_FLOAT_TYPE = simpleAggType("sum", BaseType.FLOAT);
+  public static final MeasureType SUM_DOUBLE_TYPE = simpleAggType("sum", BaseType.DOUBLE);
+  public static final MeasureType MIN_BIGINT_TYPE = simpleAggType("min", BaseType.BIGINT);
+  public static final MeasureType MIN_FLOAT_TYPE = simpleAggType("min", BaseType.FLOAT);
+  public static final MeasureType MIN_DOUBLE_TYPE = simpleAggType("min", BaseType.DOUBLE);
+  public static final MeasureType MAX_BIGINT_TYPE = simpleAggType("max", BaseType.BIGINT);
+  public static final MeasureType MAX_FLOAT_TYPE = simpleAggType("max", BaseType.FLOAT);
+  public static final MeasureType MAX_DOUBLE_TYPE = simpleAggType("max", BaseType.DOUBLE);
```

Comment:
Thank you for the comment. Yes, aggregation in Calcite for MSQ is in a precarious state. Rather than have the required debate here, I've removed the type parser (and thus aggregation types) from this PR. They can be included in a future PR.